### DISCUSSION TITLE
remark.db not persistent

### DISCUSSION THREAD

**Discussion by PaulieBleeker** [umputun/remark42#944] [Q&A] [answered]
Hi,

my site works fine with remark42 and I have entered about 6 different comments for test purposes.
However, these comments are deleted when I make changes to `docker-compose.yml` and restart the container.

I checked my configuration which seems to be alright:
```
...
- STORE_TYPE=bolt
- STORE_BOLT_PATH=/mnt/data/remark42/var/db
...
volumes:
- /mnt/data/remark42:/srv/var
```
When I run `find . -name remark.db` as root in `/` I only get the following lines:
```
./var/lib/docker/overlay2/5ec1da520fc2a84376b586758e2e2ed38d5780ddf28999c1005cb63c8eb423b6/diff/mnt/data/remark42/var/db/remark.db
./var/lib/docker/overlay2/5ec1da520fc2a84376b586758e2e2ed38d5780ddf28999c1005cb63c8eb423b6/merged/mnt/data/remark42/var/db/remark.db
```
Which looks weird to me.
I would expect to have `remark42.db` in `/mnt/data/remark42/var/db`, especially when there are already 6 comments.

With `docker-compose logs -f` I get no errors, just some info messages:
```
remark42    | 2021/04/05 19:33:16.782 [INFO]  make data store, type=bolt
remark42    | 2021/04/05 19:33:16.782 [INFO]  bolt store for sites [{FileName:/mnt/data/remark42/var/db/remark.db SiteID:remark}], options {Timeout:30s NoGrowSync:false NoFreelistSync:false FreelistType: ReadOnly:false MmapFlags:0 InitialMmapSize:0 PageSize:0 NoSync:false OpenFile:<nil>}
remark42    | 2021/04/05 19:33:16.794 [INFO]  make admin store, type=shared
remark42    | 2021/04/05 19:33:16.795 [INFO]  make cache, type=mem
remark42    | 2021/04/05 19:33:16.795 [INFO]  make avatar store, type=fs
remark42    | 2021/04/05 19:33:16.795 [INFO]  anonymous access enabled
remark42    | 2021/04/05 19:33:16.795 [INFO]  activate http rest server on port 8080
remark42    | 2021/04/05 19:33:16.795 [INFO]  activate auto-backup for remark under /mnt/data/remark42/backup, duration 24h0m0s
remark42    | 2021/04/05 19:33:16.795 [INFO]  start pictures cleanup, staging ttl=50m0s
remark42    | 2021/04/05 19:33:16.796 [INFO]  run file server for ./web, path /web
...
```
So maybe it has something to do with file permissions or my user.
In my `docker-compose.yml` I specified `APP_UID=0`. Without this value, remark42 won't show up on my site (again: no error-messages). I start the container with a non-root user with the sudo privilege.

Any I idea how to fix this (or where to look at)?

**Comment by paskal** [comment, marked as answer]
The first l question is how do you restart the container? Description of which commands you used would be very useful.

**Reply by umputun** [reply]
this part is incorrect in your compose:

`- STORE_BOLT_PATH=/mnt/data/remark42/var/db`

The path you should set is for the container, not the host FS (as you did). So, you should have should be smth like `- STORE_BOLT_PATH=/srv/var/db`

also, see [provided example](https://github.com/umputun/remark42/blob/master/docker-compose.yml) for a similar mapping

_P.S. pls note, this is not remark42 specific, but rather docker-compose specific_

**Reply by PaulieBleeker** [reply]
You nailed it. Thanks man. 👍
